docs/markdown/Fs-module.md

```diff
@@ -216,6 +215,51 @@ fs.stem('foo/bar/baz.dll.a')  # baz.dll
    project. If the file specified by `path` is a `files()` object it
    cannot refer to a built file.
 
+### relative_to
+
+*since 0.64.0*
+
+Given two paths, returns a version of the first path relative to the second.
+
+Examples: 
+
+```meson
+# On windows:
+fs.relative_to('c:\\proj1\\foo', 'c:\\proj1\\bar')  # '..\\foo'
+fs.relative_to('c:\\proj1\\foo', 'd:\\proj1\\bar')  # ERROR, since the relative path does not exist
+
+# On other systems:
+fs.relative_to('/prefix/lib', '/prefix/bin')  # '../lib'
+fs.relative_to('/prefix/lib/foo', '/prefix')  # 'lib/foo'
+```
+
+`relative_to` has some optional keyword arguments:
+
+- `native`: Can be set to `true` or `false`. Whether the relative paths are computed
+  following the rules of the build machine (native: `true`) or the host (native: `false`).
+
+- `allow_absolute`: If set to `true`, instead of giving an error, `relative_to()`
+  will return the original path. Useful if an absolute path is a reasonable
+  fallback.
+
+- `if_within`: A path. `relative_to(path1, path2, within: path3)` returns
+  the relative path of `path1 with respect to `path2` if `path1` is found in `path3`,
+  otherwise it returns `path1` unchanged.
+
+Examples: 
+
+```meson
+# On windows:
+fs.relative_to('c:\\proj1\\foo', 'c:\\proj2\\bar', if_within: 'c:\\proj2')  # ERROR, the relative path is outside the "within" argument
```

Comment:
Is that an error? Doc above says " if `path1` is found in `path3`, otherwise it returns `path1` unchanged", so shouldn't this return `c:\\proj1\\foo`?

Reply:
The example was right, the docs were wrong.

If `path1` is not found on `path3`, one can do two things: either we return `path1` or we error.

- Choosing to always error is the easy path, but it is not convenient to users because as far as I know meson doesn't have a try/catch.
- Returning `path1` as was documented may lead to confusion if both `path1` and `path2` happen to be relative paths. In that scenario we could not always know if the returned path is just `path1` we gave because it wasn't contained in `path3` or if it was relative to `path2` because it was contained in `path3`. Therefore the only safe behaviour is to return `path1` if it is absolute, or raise an error. But for consistency, absolute paths can only be returned if `allow_absolute : true`, so that needs to be set as well.

I've pushed a commit to clarify the documentation as well